airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -201,6 +201,12 @@ def ti_update_state(
 
     if isinstance(ti_patch_payload, TITerminalStatePayload):
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
+        query = query.values(
+            state=ti_patch_payload.state,
+        )
+        if ti_patch_payload.state == State.FAILED:
+            # clear the next_method and next_kwargs
+            query = query.values(next_method=None, next_kwargs=None)
```

Comment:
What about other things that we need to support, the one's linked below:

https://github.com/apache/airflow/blob/8848d6e35f6ead2cc1ed9887af8360fbd61c78c3/airflow/models/taskinstance.py#L3039-L3040

https://github.com/apache/airflow/blob/8848d6e35f6ead2cc1ed9887af8360fbd61c78c3/airflow/models/taskinstance.py#L3085-L3086

Reply:
ok I see in description: https://github.com/apache/airflow/issues/44951

Reply:
end date and duration remains